### PR TITLE
AWS - Access Advisor

### PR DESCRIPTION
**Pull request by amshamah419**
## Original External PR
[external pull request](https://github.com/demisto/content/pull/5067)

## Status
In Progress

## Related Issues
fixes: Internal for Re:Invent

## Description
AWS Access Analyzer Integration

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)
- [ ] [CHANGELOG](link)

### CODE REVIEW

File: Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    kwargs = {
+        'analyzerArn': args.get('analyzerArn'),
+        'resourceArn': args.get('resourceArn')
+    }
+
+    response = client.get_analyzed_resource(**kwargs)
+    data = json.loads(json.dumps(response['resource'], cls=DatetimeEncoder))
+    data['analyzerArn'] = args.get('analyzerArn')
+
+    ec = {'AWS.AccessAnalyzer.Analyzers(val.analyzerArn === obj.analyzerArn).Resource(val.resourceArn === obj.resourceArn)': data}
```
Comment (yuvalbenshalom) [inline Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py]: are you sure about this DT? dose it merge properly?
isn't `AWS.AccessAnalyzer.Analyzers.Resource(val.resourceArn === obj.resourceArn)` enough ?

File: Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+    client.update_findings(**kwargs)
+    return_outputs("Findings updated")
+
+
+def test_function():
+    try:
+        client = aws_session()
+        response = client.list_analyzers()
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
```
Comment (yuvalbenshalom) [inline Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py]: if yo do not get ok, better print the status code and the reason.

File: Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+            return_outputs('ok')
+
+    except Exception as e:
+        return return_error(str(e))
+
+
+"""EXECUTION BLOCK"""
+try:
```
Comment (yuvalbenshalom) [inline Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.py]: a general comment:
please change the command functions names that are called under this to `<name>_command()`

**Comment by amshamah419** [comment]
@kirbles19 Can you review these - 

https://github.com/demisto/content/blob/AccessAdvisor/Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer.yml

https://github.com/demisto/content/blob/AccessAdvisor/Beta_Integrations/AWS-AccessAnalyzer/AWS-AccessAnalyzer_description.md

**Comment by kirbles19** [comment]
@amshamah419 Done.

**Pull request merged**